src/partially_blindrsa.ts

```diff
@@ -0,0 +1,489 @@
+        const { jwkKey } = await this.extractKeyParams(publicKey, 'public');
+        if (!jwkKey.n) {
+            throw new Error('key has invalid parameters');
+        }
+        const e = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.e));
+        const n = sjcl.bn.fromBits(sjcl.codec.base64url.toBits(jwkKey.n));
+        const pk: BigPublicKey = { e, n };
+
+        // 1. Compute pk_derived = DerivePublicKey(pk, info).
+        const pk_derived = await this.derivePublicKey(pk, info);
+        const pk_derived_key = await crypto.subtle.importKey(
+            'jwk',
+            {
+                ...jwkKey,
+                e: sjcl.codec.base64url.fromBits(pk_derived.e.toBits(0)),
+                n: sjcl.codec.base64url.fromBits(pk_derived.n.toBits(0)),
+            },
+            { name: PartiallyBlindRSA.NAME, hash: this.params.hash },
+            false,
+            ['verify'],
+        );
+
+        // 2. Compute msg_prime = concat("msg", int_to_bytes(len(info), 4), info, msg).
+        const msg_prime = joinAll([
+            new TextEncoder().encode('msg'),
+            int_to_bytes(info.length, 4),
+            info,
+            message,
+        ]);
+
+        // 3. Invoke and output the result of RSASSA-PSS-VERIFY (Section 8.1.2 of [RFC8017]) with (n, e) as pk_derived, M as msg_prime, and S as sig.
+        return crypto.subtle.verify(
+            { name: PartiallyBlindRSA.NAME, saltLength: this.params.saltLength },
+            pk_derived_key,
+            signature,
+            msg_prime,
+        );
+    }
+
+    private async derivePublicKey({ n }: BigPublicKey, info: Uint8Array): Promise<BigPublicKey> {
+        // 1. hkdf_input = concat("key", info, 0x00)
+        // 2. hkdf_salt = int_to_bytes(n, modulus_len)
+        // 3. lambda_len = modulus_len / 2
+        // 4. hkdf_len = lambda_len + 16
+        const hkdf_input = joinAll([new TextEncoder().encode('key'), info, new Uint8Array([0x00])]);
+        const hkdf_salt = i2osp(n, n.bitLength() / 8);
+        const lambda_len = n.bitLength() / 8 / 2;
+        const hkdf_len = lambda_len + 16;
+
+        // 5. expanded_bytes = HKDF(IKM=hkdf_input, salt=hkdf_salt, info="PBRSA", L=hkdf_len)
+        const expanded_bytes = new Uint8Array(
+            await crypto.subtle.deriveBits(
+                {
+                    name: 'HKDF',
+                    hash: this.params.hash,
+                    info: new TextEncoder().encode('PBRSA'),
+                    salt: hkdf_salt,
+                },
+                await crypto.subtle.importKey('raw', hkdf_input, 'HKDF', false, ['deriveBits']),
+                hkdf_len * 8,
+            ),
+        );
+
+        // 6. expanded_bytes[0] &= 0x3F // Clear two-most top bits
+        // 7. expanded_bytes[lambda_len-1] |= 0x01 // Set bottom-most bit
+        expanded_bytes[0] &= 0x3f;
+        expanded_bytes[lambda_len - 1] |= 0x01;
+
+        // 8. e' = bytes_to_int(slice(expanded_bytes, lambda_len))
```

Comment:
this may be an error in the spec, because `slice` receives 3 parameters.
also, it's not clear why it requests additionally +16 bytes to the HDKF and they are not used.

Reply:
I agree. I've put a PR on the spec to correct this https://github.com/chris-wood/draft-amjad-cfrg-partially-blind-rsa/pull/49. This comment is a direct copy paste of draft 2 which is the latest spec atm

Reply:
I've updated it to `0, `, which is the same comment I made on the PR to the draft.